src/python/bot/tasks/fuzz_task.py

```diff
@@ -1319,6 +1173,156 @@ def sync_new_corpus_files(self):
 
     self.gcs_corpus.upload_files(new_files)
 
+  def generate_blackbox_testcases(self, fuzzer, fuzzer_directory,
+                                  testcase_count):
+    """Run the blackbox fuzzer and generate testcases."""
+    # Helper variables.
+    error_occurred = False
+    fuzzer_revision = fuzzer.revision
+    fuzzer_name = fuzzer.name
+    sync_corpus_directory = None
+
+    # Clear existing testcases (only if past task failed).
+    testcase_directories = get_testcase_directories(self.testcase_directory,
+                                                    self.data_directory)
+    testcase_manager.remove_testcases_from_directories(testcase_directories)
+
+    # Set an environment variable for fuzzer name.
+    # TODO(ochang): Investigate removing this. Only users appear to be chromebot
+    # fuzzer and fuzzer_logs, both of which can be removed.
+    environment.set_value('FUZZER_NAME', fuzzer_name)
+
+    # Set minimum redzone size, do not detect leaks and zero out the
+    # quarantine size before running the fuzzer.
+    environment.reset_current_memory_tool_options(
+        redzone_size=16, leaks=False, quarantine_size_mb=0)
+
+    if fuzzer.builtin:
+      fuzzer_command = 'builtin'
+      builtin_fuzzer = builtin_fuzzers.get(fuzzer.name)
+
+      builtin_result = builtin_fuzzer.run(
```

Comment:
Why is the engine/builtin code inside generate_blackbox_testcases, was this expected or no other choice.

Reply:
This will be removed once the refactor is done and we've migrated to new engine fuzzing code.